### ISSUE TITLE
Manual mesh fails to render if the vertex buffers contain doubles instead of floats

### ISSUE THREAD

**Issue by KABoissonneault** [OGRECave/ogre-next#114]
### System Information
- Ogre Version: 2.2 (commit e1e11560c988b508984e01a7958814e0f69d5bbe)
- Operating System / Platform: Windows
- RenderSystem: Direct3D11
- GPU: GTX970

#### Detailled description
I attempted to reproduce the sample "Sample_DynamicGeometry" in my code, however my project uses doubles for positions and normals, so I tried using `VET_DOUBLE3` instead of `VET_FLOAT3` for the `VES_POSITION` and `VES_NORMAL` data while creating the vertex buffers (ie: `VaoManager::createVertexBuffer`). 
However, nothing would render for static geometry, and dynamic geometry would just fly all over.
It's possible there's simply a bug somewhere in the Direct3D11 RenderSystem.

If doubles are not supported for such data, the RenderSystem should probably assert or throw when receiving such data as input

**Comment by darksylinc** [comment]
> If doubles are not supported for such data, the RenderSystem should probably assert or throw when receiving such data as input

Yes, you're right. D3D11 does not support doubles and we should raise an exception

**Comment by lingfors** [comment]
I thought that Ogre convert doubles to floats as needed if Ogre::Real is typedef'd to double.

**Comment by darksylinc** [comment]
> Sorry, I think I misunderstood the issue.

Indeed. One thing is to have Real typedef to double (which affects CPU calculations), and another one is to request VET_DOUBLEx (which affects GPU calculations but support is scarce, if any).
